auth/CognitoSecureStorage.ts

```diff
@@ -36,109 +37,121 @@ const STORE_CHUNK_SIZE = 2048
 /**
  * Encrypted secure storage for AWS Cognito Auth.
  */
-export class CognitoSecureStorage {
-  private syncPromise?: Promise<void>
+export class CognitoSecureStorage implements KeyValueStorageInterface {
   private store: SecureStore
   private keyChunkMappings = new Map<string, number>()
-  private cache = new Map<string, string>()
+  private keyChunkMappingsPromise?: Promise<Map<string, number>>
 
   constructor (store: SecureStore) {
     this.store = store
+    this.keyChunkMappingsPromise = undefined
   }
 
-  setItem (key: string, value: string) {
-    const chunks = stringToChunks(value)
-    this.keyChunkMappings.set(key, chunks.length)
-    this.saveKeyChunkMappings()
-    this.cache.set(key, value)
-    for (let i = 0; i < chunks.length; i++) {
-      this.store.setItemAsync(secureStoreKey(key, i), chunks[i])
+  async removeItem (key: string) {
+    const keyChunkMappings = await this.loadKeyChunkMappings()
+    if (keyChunkMappings) {
```

Comment:
I think `loadKeyChunkMappings` should be written such that it never returns undefined so that you don't have to do this dance in each one of the get/set/remove methods.